src/data/fin/basic.lean

```diff
@@ -777,6 +777,10 @@ ext rfl
   cast h (cast_add m' i) = cast_add m i :=
 ext rfl
 
+lemma cast_add_cast_add {m n p : ℕ} (i : fin m) :
+  cast_add p (cast_add n i) = cast (add_assoc _ _ _).symm (cast_add (n + p) i) :=
```

Comment:
I like to make proof terms in rewrite lemmas, like `(add_assoc _ _ _).symm` here, into optional parameters so that rewriting right-to-left also works. But that might be superstition since proofs are just supposed to unify anyway, right?

Reply:
I'm increasingly of the believe that that's superstition; I certainly have the same vague memory of this mattering, but the last few times it came up no one was able to produce an example where it mattered; and certainly it never solidified into a library note! So I'd prefer to just leave this as is.